webserver/src/main/java/io/kestra/webserver/controllers/api/NamespaceFileController.java

```diff
@@ -183,8 +181,12 @@ public synchronized int available() {
 
     private void putNamespaceFile(String tenantId, String namespace, URI path, BufferedInputStream inputStream) throws IOException {
         String filePath = path.getPath();
-        if(filePath.matches("/" + FLOWS_FOLDER + "/.*")) {
-            if(filePath.split("/").length != 3) {
+        if (filePath.startsWith(DELIMITER_SLASH)) {
+            filePath = filePath.substring(1);
+        }
+        String[] filePaths = filePath.split(DELIMITER_SLASH);
+        if (filePaths.length != 0 && filePaths[0].startsWith(FLOWS_FOLDER)) {
```

Comment:
```suggestion
        if (filePaths.length != 0 && filePaths[0].equals(FLOWS_FOLDER)) {
```
the startsWith was only if we compared the full path but here we should only do the import flow behaviour if the path part is actually the entire flow folder name